specification/paths/Contracts-v1-templates.json

```diff
@@ -0,0 +1,177 @@
+                "data": {
+                  "type": "array",
+                  "items": {
+                    "$ref": "#/components/schemas/ContractTemplateResponse"
+                  }
+                },
+                "meta": {
+                  "$ref": "#/components/schemas/PaginationMeta"
+                },
+                "included": {
+                  "$ref": "#/components/schemas/ContractTemplateIncludes"
+                },
+                "links": {
+                  "type": "object",
+                  "required": [
+                    "self",
+                    "first",
+                    "last"
+                  ],
+                  "additionalProperties": false,
+                  "properties": {
+                    "self": {
+                      "type": "string",
+                      "example": "$API_HOST/contracts/v1/templates?page[number]=3&page[size]=30"
+                    },
+                    "first": {
+                      "type": "string",
+                      "example": "$API_HOST/contracts/v1/templates?page[number]=1&page[size]=30"
+                    },
+                    "prev": {
+                      "type": "string",
+                      "example": "$API_HOST/contracts/v1/templates?page[number]=2&page[size]=30"
+                    },
+                    "next": {
+                      "type": "string",
+                      "example": "$API_HOST/contracts/v1/templates?page[number]=4&page[size]=30"
+                    },
+                    "last": {
+                      "type": "string",
+                      "example": "$API_HOST/contracts/v1/templates?page[number]=13&page[size]=30"
+                    }
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  },
+  "post": {
+    "tags": [
+      "ContractTemplates"
+    ],
+    "security": [
+      {
+        "OAuth2": [
+          "contract_templates.manage"
+        ]
+      }
+    ],
+    "summary": "Create a new contract template.",
+    "description": "This endpoint creates a new contract template with the posted data.",
+    "requestBody": {
+      "description": "The contract template object to be created.",
+      "required": true,
+      "content": {
+        "application/vnd.api+json": {
+          "schema": {
+            "type": "object",
+            "required": [
+              "data"
+            ],
+            "additionalProperties": false,
+            "properties": {
+              "data": {
+                "allOf": [
+                  {
+                    "$ref": "#/components/schemas/ContractTemplate"
+                  },
+                  {
+                    "required": [
+                      "attributes",
+                      "relationships"
+                    ],
+                    "properties": {
+                      "id": {
+                        "readOnly": true
+                      },
```

Comment:
this feels more like something that would go on a PUT/PATCH request

Reply:
@yoerriwalstra The resource schema we put in the `allOf` has a `type` and `id` and we have to define this readOnly to disallow posting the `id`.

It would be nice to find a way to avoid defining this readOnly, but if we don't include the `id` in the resource schema, then we would have to define it in a different way. Might be worth looking into.